### PR TITLE
Add `--safe-options`

### PR DESCRIPTION
**Pull request by ajreynol**
This option indicates that further options cannot be expert.

Note that `--safe-options --expert-option` throws an exception  while `--expert-option --safe-options` does not.

### CODE REVIEW

File: src/main/driver_unified.cpp [hunk 1]
```diff
@@ -130,15 +130,15 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
   {
     if( inputFromStdin ) {
       // We can't do any fancy detection on stdin
-      solver->setOption("input-language", "smt2");
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:133]: Should this really use setOptionInternal if `input-language` is a common option?

File: src/main/driver_unified.cpp [hunk 2]
```diff
@@ -130,15 +130,15 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
...
+      pExecutor->setOptionInternal("input-language", "smt2");
     } else {
       size_t len = filenameStr.size();
       if(len >= 5 && !strcmp(".smt2", filename + len - 5)) {
-        solver->setOption("input-language", "smt2");
+        pExecutor->setOptionInternal("input-language", "smt2");
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:137]: Ditto.

File: src/main/driver_unified.cpp [hunk 3]
```diff
@@ -130,15 +130,15 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
...
       } else if((len >= 3 && !strcmp(".sy", filename + len - 3))
                 || (len >= 3 && !strcmp(".sl", filename + len - 3))) {
         // version 2 sygus is the default
-        solver->setOption("input-language", "sygus2");
+        pExecutor->setOptionInternal("input-language", "sygus2");
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:141]: Ditto.

File: src/main/driver_unified.cpp [hunk 4]
```diff
@@ -148,7 +148,7 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
     // to simplify checking at the API level. In particular, the sygus
     // option is the authority on whether sygus commands are currently
     // allowed in the API.
-    solver->setOption("sygus", "true");
+    pExecutor->setOptionInternal("sygus", "true");
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:151]: Ditto (regular).

File: src/main/driver_unified.cpp [hunk 5]
```diff
@@ -158,7 +158,8 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
 
   if (solver->getOption("output-language") == "LANG_AUTO")
   {
-    solver->setOption("output-language", solver->getOption("input-language"));
+    pExecutor->setOptionInternal("output-language",
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:161]: Ditto.

File: src/main/driver_unified.cpp [hunk 6]
```diff
@@ -185,7 +186,8 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
       // set incremental if we are in interactive mode
       if (!solver->getOptionInfo("incremental").setByUser)
       {
-        solver->setOption("incremental", isInteractive ? "true" : "false");
+        pExecutor->setOptionInternal("incremental",
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:189]: Ditto.

File: src/main/driver_unified.cpp [hunk 7]
```diff
@@ -218,13 +220,13 @@ int runCvc5(int argc, char* argv[], std::unique_ptr<cvc5::Solver>& solver)
     {
       if (!solver->getOptionInfo("incremental").setByUser)
       {
-        solver->setOption("incremental", "false");
+        pExecutor->setOptionInternal("incremental", "false");
```
Comment (mpreiner) [inline src/main/driver_unified.cpp:223]: Ditto.

File: src/main/driver_unified.cpp (hunk 1 above)
Comment (ajreynol) [reply src/main/driver_unified.cpp:133]: The point is to distinguish options that the user set vs. those that are set automatically.

In my opinion, it is cleaner to consider *all* options setting that main does in the same way, regardless of the category of the options. This was `fromUser` in `SolverEngine::setOption` really means "the user intentionally set this option".

**Review by mpreiner — APPROVED** [review]

**Pull request merged**